training/src/anemoi/training/schemas/dataloader.py

```diff
@@ -57,10 +59,10 @@ def as_seconds(self) -> int:
         return int(self.as_timedelta.total_seconds())
 
 
-class DatasetSchema(BaseModel):
+class DatasetSchema(PydanticBaseModel):
     """Dataset configuration schema."""
 
-    dataset: Union[str, dict, Path, list[dict]]
+    dataset: Union[str, dict, Path, list[dict]] | None = None
```

Comment:
I think you might need to update this type hint to avoid using | so unit tests can pass with 3.9, something like  `Union[Union[str, dict, Path, list[dict]],None]` should work